transaction/core/src/mint/tx.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Minting transactions.
+
+use crate::domain_separators::MINT_TX_PREFIX_DOMAIN_TAG;
+use alloc::vec::Vec;
+use core::fmt;
+use mc_crypto_digestible::{Digestible, MerlinTranscript};
+use mc_crypto_keys::{Ed25519Signature, RistrettoPublic};
+use mc_crypto_multisig::MultiSig;
+use mc_util_serial::Message;
+use serde::{Deserialize, Serialize};
+
+/// The contents of a mint-tx, which is a transaction to mint new tokens.
+#[derive(
+    Clone, Deserialize, Digestible, Eq, Hash, Message, Ord, PartialEq, PartialOrd, Serialize,
+)]
+pub struct MintTxPrefix {
+    /// Token ID we are minting.
+    #[prost(uint32, tag = "1")]
+    pub token_id: u32,
+
+    /// Amount we are minting.
+    #[prost(uint64, tag = "2")]
+    pub amount: u64,
+
+    /// The destination's public subaddress view key 'C'.
+    #[prost(message, required, tag = "3")]
+    pub view_public_key: RistrettoPublic,
+
+    /// The destination's public subaddress spend key `D`.
+    #[prost(message, required, tag = "4")]
+    pub spend_public_key: RistrettoPublic,
+
+    /// Nonce, to prevent replay attacks.
```

Comment:
Nit: ideally we would mention the length of this bytes field as well in the code comment

Reply:
Will update in a different PR so that this one doesnt need to get approved again.